documentation/configuration.md

```diff
@@ -128,3 +128,139 @@ The output of command should resemble the following JSON object:
+
+```json
+{
+  "Storage": {
+    "DumpTempFolder": "/ephemeral-directory/"
+  }
+}
+```
+
+## Cross-Origin Resource Sharing (CORS) Configuration
+
+// TODO
+
+## Metrics Configuration
+
+### Metrics Urls
+
+In addition to the ordinary diagnostics urls that `dotnet-monitor` binds to, it also binds to metric urls that only expose the `/metrics` endpoint. Unlike the other endpoints, the metrics urls do not require authentication. Unless you enable collection of custom providers that may contain sensitive business logic, it is generally consider safe to expose metrics endpoints. 
+
+Metrics urls can configured via the command line:
+
+```cmd
+dotnet monitor collect --metricUrls http://*:52325/
+```
+
+Or configured via a configuration file:
+
+```json
+{
+  "Metrics": {
+    "Endpoints": "http://localhost:52325"
+  }
+}
+```
+
+### Customize collection interval and counts
+
+In the default configuration, `dotnet-monitor` requests that the connected runtime provides updated counter values every 10 seconds and will retain 3 data point for every collected metric. When using a collection tool like Prometheus, it is recommended that you set your scrape interval to `MetricCount` * `UpdateIntervalSeconds`. In the default configuration, we recommend you scrape `dotnet-monitor` for metrics every 30 seconds.
+
+You can customize the number of data points stored per metric and the frequency at which the runtime updates each metric via the following configuration:
+
+```json
+{
+  "Metrics": {
+    "MetricCount": 3,
+    "UpdateIntervalSeconds": 10,
+  }
+}
+```
+
+### Custom Metrics
+
+Additional metrics providers and counter names to return from this route can be specified via configuration. 
+
+```json
+{
+  "Metrics": {
+    "Providers": [
+      {
+        "ProviderName": "Microsoft-AspNetCore-Server-Kestrel",
+        "CounterNames": [
+          "connections-per-second",
+          "total-connections"
+        ]
+      }
+    ]
+  }
+}
+```
+
+#### Allow insecure channel for custom metrics
+
+In the default configuration, enabling of custom metrics changes the default binding address from `http` to `https` for the metrics urls. To continue egressing custom metric via an endpoint with `http` scheme, you explicit opt-in to it. You can do so via the following configuration:
+
+```json
+{
+  "Metrics": {
+    "AllowInsecureChannelForCustomMetrics": true
+  }
+}
+```
+
+### Disable default providers
+
+In addition to enabling custom providers, `dotnet-monitor` also allows you to disable to collection of the default providers. You can do so via the following configuration:
+
+```json
+{
+  "Metrics": {
+    "IncludeDefaultProviders": false
+  }
+}
+```
+
+## Egress Configuration
+
+// TODO
```

Comment:
Fyi, I have descriptions and samples in a separate changeset for egress. I can move them here or you can link from here to definitions.

Reply:
Awesome! I was hoping you've already written these 😄

Let's move them here?